src/components/createContainer.jsx

```diff
@@ -0,0 +1,108 @@
+/* @flow */
+
+import React, { Component, PropTypes } from 'react';
+import invariant from 'invariant';
+
+import getDisplayName from '../utils/getDisplayName';
+import createAdaptorShape from '../utils/createAdaptorShape';
+
+const adaptorShape = createAdaptorShape(PropTypes);
+
+export default function createContainer(DecoratedComponent, specs) {
```

Comment:
you seem to have rolled back my differentiation between 'initialArgs' (which requires containers to explicitly call 'setArgs()' to trigger requirement changes) and 'args' (which makes args a pure function of props, and the updating and refreshing handled inside this container.

```
componentWillUpdate(nextProps){
  if(this.props != nextProps){
    this.fetch(specs.args(nextProps));
  }
}
```